Sources/KIF/Additions/UITouch-KIFAdditions.m

```diff
@@ -117,6 +118,12 @@ - (void)setPhaseAndUpdateTimestamp:(UITouchPhase)phase
     [self setPhase:phase];
 }
 
+- (void)setIsFromEdge:(BOOL)isFromEdge
+{
+    NSInteger edgeType = isFromEdge ? 4 : 0;
+    [self _setEdgeType:edgeType];
+}
```

Comment:
What would happen if we directly called `_UITouchSetBasicTouchPropertiesFromEvent`? Does Apple already have the appropriate heuristics to set this properly?

Reply:
Apple derives the value of the property from the HIDEvent. We would need to synthesize a different HIDEvent for edge pan before calling `_UITouchSetBasicTouchPropertiesFromEvent`. This introduces additional complexity and I lack the necessary knowledge to implement it.

Reply:
I wonder if we could replace these magic numbers with Apple enums.